packages/html/src/datepicker/datepicker.spec.tsx

```diff
@@ -0,0 +1,117 @@
+import { classNames, States, Size, Roundness, FillMode } from '../utils';
+import {
+    Input,
+    InputPrefix,
+    InputSuffix,
+    InputInnerInput,
+    InputClearValue,
+    InputLoadingIcon,
+    InputValidationIcon
+} from '../input';
+import { Button } from '../button';
+import { Popup } from '../popup';
+import { Calendar } from '../calendar';
+
+export const DATEPICKER_CLASSNAME = `k-datepicker`;
+
+const states = [
+    States.hover,
+    States.focus,
+    States.valid,
+    States.invalid,
+    States.required,
+    States.disabled,
+    States.loading
+];
+
+const options = {
+    size: [ Size.small, Size.medium, Size.large ],
+    rounded: [ Roundness.small, Roundness.medium, Roundness.large, Roundness.full ],
+    fillMode: [ FillMode.solid, FillMode.flat, FillMode.outline ],
+};
+
+export type KendoDatePickerOptions = {
+    size?: (typeof options.size)[number] | null;
+    rounded?: (typeof options.rounded)[number] | null;
+    fillMode?: (typeof options.fillMode)[number] | null;
+};
+
+export type KendoDatePickerProps = KendoDatePickerOptions & {
+    prefix?: JSX.Element;
+    suffix?: JSX.Element;
+    value?: string;
+    placeholder?: string;
+    opened?: boolean;
+};
+
+export type KendoDatePickerState = { [K in (typeof states)[number]]?: boolean };
+
+export const DatePicker = (
+    props: KendoDatePickerProps &
+        KendoDatePickerState &
+        React.HTMLAttributes<HTMLSpanElement>
+) => {
+    const {
+        prefix,
+        suffix,
+        value,
+        placeholder,
+        size,
+        rounded,
+        fillMode,
+        hover,
+        focus,
+        valid,
+        invalid,
+        required,
+        loading,
+        disabled,
+        opened,
+        ...other
+    } = props;
+
+
+    return (
+        <>
+            <Input
+                {...other}
+                size={size}
+                rounded={rounded}
+                fillMode={fillMode}
+                hover={hover}
+                focus={focus}
+                valid={valid}
+                invalid={invalid}
+                required={required}
+                loading={loading}
+                disabled={disabled}
+                className={classNames(props.className, DATEPICKER_CLASSNAME)}
+            >
+                <InputPrefix>{prefix}</InputPrefix>
+                <InputInnerInput placeholder={placeholder} value={value} />
+                <InputValidationIcon {...props} />
+                <InputLoadingIcon {...props} />
+                <InputClearValue {...props} />
+                <InputSuffix>{suffix}</InputSuffix>
+                <Button
+                    className="k-input-button"
+                    icon="calendar"
+                    rounded={null}
+                    size={size}
+                    fillMode={fillMode}
+                />
```

Comment:
Can we filter the properties which are passed to child component? One of the benefits of the spec would be to give a clear flow of customization properties.

We can stick to the option properties only, as they are the ones that bring value to reading the spec. (e.g. answering the question "if the grid has size='small' should all dropdowns inside have their size modifeid").

I believe this is present on multiple occasions and is not specific to the DatePicker spec only.